templates/workspace_services/guacamole/guacamole-server/guacamole-auth-azure/src/main/java/org/apache/guacamole/auth/azuretre/connection/ConnectionService.java

```diff
@@ -88,64 +102,42 @@
         return configs;
     }
 
-    private JSONArray getVMsFromProjectAPI(final AzureTREAuthenticatedUser user) {
-
+    private JSONArray getVMsFromProjectAPI(final AzureTREAuthenticatedUser user) throws GuacamoleException {
         final JSONArray virtualMachines;
-
-        // Todo: Implement / Uncomment when the relevant API call is available for consumption
-        // https://github.com/microsoft/AzureTRE/issues/558
-    /*
-    try {
-      SSLContextBuilder builder = new SSLContextBuilder();
-      builder.loadTrustMaterial(null, new TrustSelfSignedStrategy());
-      SSLConnectionSocketFactory sslsf = new SSLConnectionSocketFactory(builder.build());
-      CloseableHttpClient httpclient = HttpClients.custom().setSSLSocketFactory(sslsf).build();
-
-      try {
-
-        URI projectUri = new URI(System.getenv("PROJECT_URL"));
-
-        // specify the host, protocol, and port
-        URIBuilder uriBuilder = new URIBuilder();
-
-        uriBuilder.setScheme(projectUri.getScheme()).setHost(projectUri.getHost()).setPort(projectUri.getPort())
-            .setPath("/api/userserviceinstances").setParameter("ResourceGroupId", System.getenv("RESOURCE_GROUP_ID"))
-            .setParameter("ResourceType", "virtual-desktop-guacamole");
-
-        URI uri = uriBuilder.build();
-        HttpGet httpget = new HttpGet(uri);
-        httpget.addHeader("Authorization", "Bearer " + user.getAccessToken());
-
-        CloseableHttpResponse httpResponse = httpclient.execute(httpget);
-
-        String json = EntityUtils.toString(httpResponse.getEntity());
-        if (json.length() != 0) {
-          virtualMachines = new JSONArray(json);
-        } else {
-          virtualMachines = new JSONArray();
+        try {
+            SSLContextBuilder builder = new SSLContextBuilder();
+            builder.loadTrustMaterial(null, new TrustSelfSignedStrategy());
+            SSLConnectionSocketFactory sslsf = new SSLConnectionSocketFactory(builder.build());
```

Comment:
Is it required to adjust the SSL things to make a secure HTTPS call?

Reply:
Following the next comment this was deleted.